plugins/gcov/lib/gcov_constants.rb

```diff
@@ -8,10 +8,15 @@
 GCOV_RESULTS_PATH               = File.join(GCOV_BUILD_PATH, "results")
 GCOV_DEPENDENCIES_PATH          = File.join(GCOV_BUILD_PATH, "dependencies")
 GCOV_ARTIFACTS_PATH             = File.join(PROJECT_BUILD_ARTIFACTS_ROOT, GCOV_ROOT_NAME)
+GCOV_REPORT_GENERATOR_PATH      = File.join(GCOV_ARTIFACTS_PATH, "ReportGenerator")
 
 GCOV_ARTIFACTS_FILE_HTML        = File.join(GCOV_ARTIFACTS_PATH, "GcovCoverageResults.html")
 GCOV_ARTIFACTS_FILE_COBERTURA   = File.join(GCOV_ARTIFACTS_PATH, "GcovCoverageCobertura.xml")
 GCOV_ARTIFACTS_FILE_SONARQUBE   = File.join(GCOV_ARTIFACTS_PATH, "GcovCoverageSonarQube.xml")
 GCOV_ARTIFACTS_FILE_JSON        = File.join(GCOV_ARTIFACTS_PATH, "GcovCoverage.json")
 
+# gcovr supports regular expressions.
 GCOV_FILTER_EXCLUDE             = '^vendor.*|^build.*|^test.*|^lib.*'
+
+# Report Generator supports text with wildcard characters.
+GCOV_REPORT_GENERATOR_FILE_FILTERS = '-./vendor/*;-./build/*;-./test/*;-./lib/*;-.\vendor\*;-.\build\*;-.\test\*;-.\lib\*'
```

Comment:
I have noticed osify_path_separators() method in tool_executor_helper or use condition like `if (@system_wrapper.windows?)` to create only paths for linux or windows way?

Reply:
I made some changes to the way the constants are defined. I also called `osify_path_separators()` within reportgenerator_reporinator.rb to create the Windows appropriate file filters. I want to put  path separator in the filter in case someone puts vendor, build, etc in the middle of a file name. For example, I like to make builders when I follow the factory pattern in other languages.